internal/graph/check.go

```diff
@@ -519,9 +528,27 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 			}
 
 			// filter out invalid tuples yielded by the database query
-			err = validation.ValidateTuple(typesys, tk)
+			tupleKey := t.GetKey()
+			err = validation.ValidateTuple(typesys, tupleKey)
 
 			if t != nil && err == nil {
+				tupleCondition := tupleKey.GetCondition()
```

Comment:
note to self (@jon-whit ) - we can consolidate this since we use it in a couple of places